app/src/main/java/com/keylesspalace/tusky/di/AppModule.kt

```diff
@@ -64,20 +64,23 @@ class AppModule {
         return TimelineCasesImpl(api, eventHub)
     }
 
-    @Provides
-    @Singleton
-    fun providesAccountManager(app: TuskyApplication): AccountManager {
-        return app.serviceLocator.get(AccountManager::class.java)
-    }
-
     @Provides
     @Singleton
     fun providesEventHub(): EventHub = EventHubImpl
 
     @Provides
     @Singleton
-    fun providesDatabase(app: TuskyApplication): AppDatabase {
-        return app.serviceLocator.get(AppDatabase::class.java)
+    fun providesDatabase(appContext: Context): AppDatabase {
+        return Room.databaseBuilder(appContext, AppDatabase::class.java, "tuskyDB")
+                .allowMainThreadQueries()
```

Comment:
I would disallow allowMainThreadQueries back maybe but later

Reply:
YES, but we do main thread queries and that can't be fixed easily (its actually a huge problem, we have quite some app-not-responding errors because of them)

Reply:
Oh, I forgot about that. That's no-no. Maybe you can point to this place later.